File: example-java/src/main/java/ru/service/MyService.java
package ru.service;

import reactor.core.publisher.*;
import reactor.core.scheduler.*;
import ru.communicator.*;
import ru.model.*;
import static io.art.core.factory.SetFactory.*;
import static io.art.logging.LoggingModule.*;
import static ru.communicator.Communicators.*;
import java.time.*;
import java.util.*;

public class MyService implements MyClient {
    @Override
    public void myMethod1() {
        logger(MyService.class).info("myMethod1");
        while (true) {

        }
    }


    @Override
    public void myMethod2(Request request) {
        myClient().myMethod1();
        while (true) {

        }
    }

    @Override
    public void myMethod3(Mono<Request> request) {
        logger(MyService.class).info("myMethod3:" + request.block());
    }

    @Override
    public void myMethod4(Flux<Request> request) {
        logger(MyService.class).info("myMethod4:" + request.blockFirst());
    }


    @Override
    public Response myMethod5() {
        logger(MyService.class).info("myMethod5");
        return Response.builder().build();
    }

    @Override
    public Response myMethod6(Request request) {
        logger(MyService.class).info("myMethod6:" + request);
        return Response.builder().build();
    }

    @Override
    public Response myMethod7(Mono<Request> request) {
        logger(MyService.class).info("myMethod7:" + request.block());
        return Response.builder().build();
    }

    @Override
    public Response myMethod8(Flux<Request> request) {
        Request first = request.blockFirst();
        logger(MyService.class).info("myMethod8:" + first);
        return Response.builder().build();
    }


    @Override
    public Mono<Response> myMethod9() {
        logger(MyService.class).info("myMethod9");
        return Mono.just(Response.builder().build());
    }

    @Override
    public Mono<Response> myMethod10(Request request) {
        logger(MyService.class).info("myMethod10:" + request);
        return Mono.just(Response.builder().build());
    }

    @Override
    public Mono<Response> myMethod11(Mono<Request> request) {
        logger(MyService.class).info("myMethod11");
        return Mono.just(Response.builder().build());
    }

    @Override
    public Mono<Response> myMethod12(Flux<Request> request) {
        logger(MyService.class).info("myMethod12");
        return Mono.just(Response.builder().build());
    }


    @Override
    public Flux<Response> myMethod13() {
        logger(MyService.class).info("myMethod13");
        return Flux.interval(Duration.ofSeconds(1), Schedulers.newParallel("myMethod13")).map(index -> Response.builder().build());
    }

    @Override
    public Flux<Response> myMethod14(Request request) {
        logger(MyService.class).info("myMethod14:" + request);
        return Flux.just(Response.builder().build());
    }

    @Override
    public Flux<Response> myMethod15(Mono<Request> request) {
        logger(MyService.class).info("myMethod15");
        return Flux.just(Response.builder().build());
    }

    @Override
    public Flux<Response> myMethod16(Flux<Request> request) {
        logger(MyService.class).info("myMethod16");
        return Flux.empty();
    }

    @Override
    public String myMethod17(String request) {
        logger(MyService.class).info("myMethod17");
        return "Mono.empty()";
    }

    @Override
    public Set<String> myMethod18(List<String> request) {
        logger(MyService.class).info("myMethod18");
        return setOf("Mono.empty()");
    }

    @Override
    public GenericModel<String, GenericTypeParameter<String>> myMethod19(GenericModel<String, GenericTypeParameter<String>> request) {
        logger(MyService.class).info("myMethod19");
        return null;
    }
}

File: example.kotlin/src/main/java/ru/service/MyService.kt
package ru.service

import io.art.kotlin.communicator
import io.art.kotlin.logger
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono
import ru.communicator.MyClient
import ru.model.Request
import ru.model.Response

object MyService : MyClient {
    override fun myMethod1() = logger().info("myMethod1")
    override fun myMethod2(request: Request) = communicator<MyClient>().myMethod6(Request()).run { logger().info("myMethod2") }
    override fun myMethod3(request: Mono<Request>) = logger().info("myMethod3:${request.block()}")
    override fun myMethod4(request: Flux<Request>) = logger().info("myMethod4:${request.blockFirst()}")
    override fun myMethod5(): Response = Response().also { logger().info("myMethod5") }
    override fun myMethod6(request: Request) = Response().also { logger().info("myMethod6") }
    override fun myMethod7(request: Mono<Request>) = Response().also { logger().info("myMethod7:${request.block()}") }
    override fun myMethod8(request: Flux<Request>) = Response().also { logger().info("myMethod8:${request.blockFirst()}") }
    override fun myMethod9(): Mono<Response> = Mono.empty<Response>().also { logger().info("myMethod9") }
    override fun myMethod10(request: Request): Mono<Response> = Mono.empty<Response>().also { logger().info("myMethod10:$request") }
    override fun myMethod11(request: Mono<Request>): Mono<Response> = Mono.empty<Response>().also { logger().info("myMethod11:${request.block()}") }
    override fun myMethod12(request: Flux<Request>): Mono<Response> = Mono.empty<Response>().also { logger().info("myMethod12:${request.blockFirst()}") }
    override fun myMethod13(): Flux<Response> = Flux.empty<Response>().also { logger().info("myMethod13") }
    override fun myMethod14(request: Request): Flux<Response> = Flux.empty<Response>().also { logger().info("myMethod14:$request") }
    override fun myMethod15(request: Mono<Request>): Flux<Response> = Flux.empty<Response>().also { logger().info("myMethod15:${request.block()}") }
    override fun myMethod16(request: Flux<Request>): Flux<Response> = Flux.empty<Response>().also { request.subscribe { data: Request -> logger().info("myMethod16:$data") } }
    override fun myMethod17(request: String) = "myMethod17:$request"
    override fun myMethod18(request: List<List<List<String>>>) = "myMethod18:$request"
    override fun myMethod19(request: List<Array<List<String>>>) = "myMethod19:$request"
}
